### DISCUSSION TITLE
Any way to not use Context API?

### DISCUSSION THREAD

**Discussion by SaavanNanavati** [ariakit/ariakit#1935] [Questions]
The use of Context API (I believe?) seems to be hitting performance a lot, according to profiling. I think it's because context causes deep re-rendering, even if nothing has changed. Is there any way to not use it when using Combobox?

**Comment by diegohaz** [comment]
We have a lot of internal optimizations to guarantee that only affected components re-render when the state changes. Anyway, I'd appreciate it if you could give more details.

**Comment by grigorii-lapshin** [comment]
I think I have found the simillar issue. 
I made a Select with Combobox for Timezones list. It has ≈500 members. Implementation is copied from select-combobox-tab-various example. When I add `render={<Ariakit.ComboboxItem/>}` to my SelectItem, I encounter a multi second delay before the popover renders.

**Reply by diegohaz** [reply]
That's likely related to https://github.com/ariakit/ariakit/issues/3451. I'll look into it very soon.

However, with 500 items combining both Select and Combobox, you'll probably want to use [virtualization](https://newsletter.ariakit.org/p/virtualization).
